debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/MongoDbReplicaSet.java

```diff
@@ -278,6 +278,15 @@ public void start() {
      */
     @Override
     public void stop() {
+        if (started) {
+            forceStop();
+        }
+    }
+
+    public void forceStop() {
+        LOGGER.info("[{}] Stopping...", name);
+        MoreStartables.deepStopSync(members.stream());
+        started = false;
```

Comment:
Can you elaborate on this change please?

Reply:
I found that there is still some cases where `Stopping [rs0] .......`  was output on the console and stop was triggered in tests where no MongoDB instance was running and fixed it in MongoDbShardedCluster. Then I realized I remove the option to force stop and fixed it there in that way so the original code is still callable via `forceStop()`. If you say it is not needed then I can revert this change here and solve it in MongoDbShardedCluster in the same way. I just tried to do it in a less intrusive way here.

Reply:
I wouldn't worry about being intrusive in this case -- it's just an internal testing tool... Though I wonder how log output would be getting into tests when there was no mongo running

Reply:
@rk3rn3r  Looking at the comment... seems I was unabele to capture the entire snippet... This is what I found strange

```
public void forceStop() {
        LOGGER.info("[{}] Stopping...", name);
        MoreStartables.deepStopSync(members.stream());
        started = false;
        if (started) {
            LOGGER.info("[{}] Stopping...", name);
            MoreStartables.deepStopSync(members.stream());
            started = false;
        }
    }
```

should the conditioned block be there?

Reply:
Maybe you like to switch to side-by-side diffs eventually? ;)

Reply:
> Though I wonder how log output would be getting into tests when there was no mongo running

For simplicity `RestExtensionTestInfrastructure#stopContainers()` iterates over all known containers and runs `#stop()` on them. This can be improved for sure.
Normally all Container implementations return directly like here with an early return, when the underlying container is not running. And they just ignore the request to stop.
That way calling `#stop()` was idempotent and could be called mutliple times. As we didn't do this here the output was generated. Furthermore imo in some cases even the request to stop the container would have been triggered by Testcontainers and that means the request timeout to Docker deamon calls applies here. Leading to a slightly longer test execution. But that's just guessed.

Reply:
I will then remove the `#forceStop` and just include the code inside the overriden `#stop()` method. Thx

Reply:
That would remove the indempotence no? Something we should keep.

Reply:
No, it keeps idempotency but removes the non-idempotent option to enforce stopping (on a non-running container).